podcasts/Onboarding/LoginCoordinator.swift

```diff
@@ -95,37 +90,31 @@ class LoginCoordinator: NSObject, OnboardingModel {
 
 // MARK: - Social Buttons
 extension LoginCoordinator {
-    func signInWithAppleTapped() {
-        let appleIDProvider = ASAuthorizationAppleIDProvider()
-        let request = appleIDProvider.createRequest()
-        request.requestedScopes = [.email]
-
-        let authorizationController = ASAuthorizationController(authorizationRequests: [request])
-        authorizationController.delegate = self
-        authorizationController.presentationContextProvider = self
-        authorizationController.performRequests()
-    }
-
     @MainActor
-    func signInWithGoogleTapped() {
+    func signIn(with provider: SocialAuthProvider) {
         guard let navigationController else {
             return
         }
 
+        socialLogin = SocialLoginFactory.provider(for: provider, from: navigationController)
+
         Task {
             progressAlert = SyncLoadingAlert()
             do {
                 // First get token
-                try await self.googleSocialLogin.getToken(from: navigationController)
+                try await self.socialLogin?.getToken()
 
                 // If token is returned, perform login on our servers
                 await withUnsafeContinuation { continuation in
                     progressAlert?.showAlert(navigationController, hasProgress: false) {
                         continuation.resume()
                     }
                 }
-                let response = try await self.googleSocialLogin.login()
-                newAccountCreated = response.isNewAccount ?? false
+
+                listenToSync()
+
+                let response = try await self.socialLogin?.login()
+                newAccountCreated = response?.isNewAccount ?? false
```

Comment:
What about moving this at the end since the refresh won't start until the next runloop, and this will ensure that we only add the listeners once the user has actually successfully signed in. 

```suggestion
                let response = try await self.socialLogin?.login()
                newAccountCreated = response?.isNewAccount ?? false
                listenToSync()
```